### PR TITLE
feat: Add changes to Snaps Address UI component

### LINKED ISSUE #2758: Address component to display user-owned accounts via their client names
Let's parse a user's own accounts to be represented via their display names (e.g. "Gaming Account 1") instead of the truncated address (e.g. "0xEqT...83x") in our accounts component.

[Figma](https://www.figma.com/design/gHWl7G8ih3YXesly2XaA0R/Snaps-DS?node-id=329-1524&t=yPhqvbM20tFUWtv8-1)

### PR DESCRIPTION
**Pull request by david0xd**
## **Description**
Add changes to Snaps Address UI component to support more flexible use cases.

This PR needs release of the `snaps-sdk` after merging: https://github.com/MetaMask/snaps/pull/2833

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27798?quickstart=1)

## **Related issues**
Fixes: https://github.com/MetaMask/snaps/issues/2758

## **Manual testing steps**
1. Use custom Snap build from template-snap-monorepo and add few varieties of address component to test new capabilities.
2. Make sure UI is looking good.
3. Make sure that there are no regressions.

### **Before**
Screenshot 2024-10-11 at 18 12 06

### **After**
Screenshot 2024-10-15 at 14 34 16
Screenshot 2024-10-15 at 14 34 28
Screenshot 2024-10-15 at 14 34 47
Screenshot 2024-10-15 at 14 35 10

#### (_Might be useful for QA_)
```typescript
export const onRpcRequest: OnRpcRequestHandler = async ({ request }) => {
  switch (request.method) {
    case 'hello':
      return snap.request({
        method: 'snap_dialog',
        params: {
          type: 'confirmation',
          content: (
            <Box>
              <Heading size="lg">Some unknown address</Heading>

              <Text>Default address</Text>
              <Address address="0x1acC6721CF7642Eb37651F8019AC9a785C064002" />

              <Text>Custom address (displayName=true)</Text>
              <Address
                address="0x1acC6721CF7642Eb37651F8019AC9a785C064002"
                displayName={true}
              />

              <Text>Custom address (avatar=false)</Text>
              <Address
                address="0x1234567890123456789012345678901234567890"
                avatar={false}
              />

              <Text>Custom address (displayName=true, avatar=false)</Text>
              <Address
                address="0x1234567890123456789012345678901234567890"
                displayName={true}
                avatar={false}
              />

              <Text>Custom address (truncate=false)</Text>
              <Address
                address="0x1234567890123456789012345678901234567890"
                truncate={false}
              />

              <Text>Custom address (truncate=false, avatar=false)</Text>
              <Address
                address="0x1234567890123456789012345678901234567890"
                truncate={false}
                avatar={false}
              />

              <Heading size="lg">My address</Heading>

              <Text>Default address</Text>
              <Address address="0xBc641D258aA0e90F70AaDbcC4dE2D58d88029713" />

              <Text>Custom address (displayName=true)</Text>
              <Address
                address="0xBc641D258aA0e90F70AaDbcC4dE2D58d88029713"
                displayName={true}
              />

              <Text>Custom address (avatar=false)</Text>
              <Address
                address="0xBc641D258aA0e90F70AaDbcC4dE2D58d88029713"
                avatar={false}
              />

              <Text>Custom address (displayName=true, avatar=false)</Text>
              <Address
                address="0xBc641D258aA0e90F70AaDbcC4dE2D58d88029713"
                displayName={true}
                avatar={false}
              />

              <Text>Custom address (truncate=false)</Text>
              <Address
                address="0xBc641D258aA0e90F70AaDbcC4dE2D58d88029713"
                truncate={false}
              />

              <Text>Custom address (truncate=false, avatar=false)</Text>
              <Address
                address="0xBc641D258aA0e90F70AaDbcC4dE2D58d88029713"
                truncate={false}
                avatar={false}
              />

              <Heading size="lg">Contact address</Heading>

              <Text>Default address</Text>
              <Address address="0x1acC6721CF7642Eb37651F8019AC9a785C064002" />

              <Text>Custom address (displayName=true)</Text>
              <Address
                address="0x1acC6721CF7642Eb37651F8019AC9a785C064002"
                displayName={true}
              />

              <Text>Custom address (avatar=false)</Text>
              <Address
                address="0x1acC6721CF7642Eb37651F8019AC9a785C064002"
                avatar={false}
              />

              <Text>Custom address (displayName=true, avatar=false)</Text>
              <Address
                address="0x1acC6721CF7642Eb37651F8019AC9a785C064002"
                displayName={true}
                avatar={false}
              />

              <Text>Custom address (truncate=false)</Text>
              <Address
                address="0x1acC6721CF7642Eb37651F8019AC9a785C064002"
                truncate={false}
              />

              <Text>Custom address (truncate=false, avatar=false)</Text>
              <Address
                address="0x1acC6721CF7642Eb37651F8019AC9a785C064002"
                truncate={false}
                avatar={false}
              />
            </Box>
          ),
        },
      });
    default:
      throw new Error('Method not found.');
  }
};
```

## **Pre-merge author checklist**
- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**
- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/snaps/snap-ui-address/snap-ui-address.tsx [hunk 1]
```diff
@@ -20,11 +21,19 @@ export type SnapUIAddressProps = {
   address: string;
   // This is not currently exposed to Snaps.
   avatarSize?: 'xs' | 'sm' | 'md' | 'lg';
+  diameter?: number;
```
Comment (hmalik88) [inline ui/components/app/snaps/snap-ui-address/snap-ui-address.tsx:24]: what's this prop for?

Comment (GuillaumeRx) [reply ui/components/app/snaps/snap-ui-address/snap-ui-address.tsx:24]: Forgot to remove it, good catch :)

File: ui/components/app/snaps/snap-ui-address/snap-ui-address.tsx [hunk 2]
```diff
@@ -20,11 +21,18 @@ export type SnapUIAddressProps = {
...
+  displayName?: boolean;
+  avatar?: boolean;
 };
 
 export const SnapUIAddress: React.FunctionComponent<SnapUIAddressProps> = ({
   address,
   avatarSize = 'md',
+
+  truncate = true,
+  displayName = false,
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-ui-address/snap-ui-address.tsx:34]: Should this be default `true`?

Comment (FrederikBolding) [inline ui/components/app/snaps/snap-ui-address/snap-ui-address.tsx:34]: In the previous PR you changed the default to show the name, is it intentional to revert that?

File: ui/components/app/snaps/snap-ui-address/snap-ui-address.tsx [hunk 3]
```diff
@@ -41,25 +49,28 @@ export const SnapUIAddress: React.FunctionComponent<SnapUIAddressProps> = ({
...
-        ? toChecksumHexAddress(parsed.address)
-        : parsed.address,
-    );
+  const formattedAddress = truncate
+    ? shortenAddress(transformedAddress)
+    : address;
 
   return (
-    <Box
-      className="snap-ui-renderer__address"
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-ui-address/snap-ui-address.tsx:56]: Can we keep the class name?

File: ui/components/app/snaps/snap-ui-address/snap-ui-address.tsx (hunk 2 above)
Comment (GuillaumeRx) [reply ui/components/app/snaps/snap-ui-address/snap-ui-address.tsx:34]: No, I think David had that to false when he did this, will default to true

Comment (GuillaumeRx) [reply ui/components/app/snaps/snap-ui-address/snap-ui-address.tsx:34]: Yes, Looking at the definition of the Address component `displayName` defaults to false

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**